src/index.js.flow

```diff
@@ -1,297 +1,80 @@
-};
-
-type StaticRouterProps = {
-  isStatic?: boolean,
-  location?: string,
-  routes: Routes,
-  children: Node,
-  resourceData?: ResourceStoreData,
-  resourceContext?: ResourceStoreContext,
-};
-
-export type MatchParams = {
-  [key: string]: string | null | typeof undefined,
-};
-
-export type ResourceStoreData =
-  | { [type: string]: RouteResourceDataForType }
-  | {||};
-
-export type RouterStoreContext = {|
-  route: Route,
-  match: Match,
-  query: Query,
-  location: Location,
-|};
-
-export type HistoryAction = 'PUSH' | 'REPLACE' | 'POP' | '';
-
-export type NavigationType = 'container' | 'product';
-
-type NavigationRenderUpdater = (
-  location: Location,
-  match: Match,
-  route: Route
-) => Node;
-
-export type NavigationStatics = {
-  [key: string]: any,
-  type: NavigationType,
-  view: (...args: any[]) => string,
-};
-
-export type Navigation = {
-  [key: string]: any,
-  ...NavigationStatics,
-  renderNavigationUpdater: NavigationRenderUpdater,
-  sidebar?: boolean,
-};
-
-export type HistoryBlocker = (
-  location: Location,
-  action: string
-) => boolean | Promise<boolean>;
-
-type RouterState = PublicStateProperties;
-
-export type RouterActionPush = (
-  path: string | LocationShape,
-  state?: any
-) => RouterAction;
-
-export type RouterActionReplace = (
-  path: string | LocationShape,
-  state?: any
-) => RouterAction;
-
-export type LinkProps = {|
-  children: Node,
-  target?: '_blank' | '_self' | '_parent' | '_top',
-  href?: string,
-  to?: string,
-  replace?: boolean,
-  type?: 'a' | 'button',
-  onClick?: (e: any) => void,
-|};
-
-export type RouterSubscriberProps = {|
-  children: (
-    state: RouterState,
-    actions: BoundActions<RouterActionsType>
-  ) => Node,
-|};
-
-// Components
+import type {
+  HistoryAction,
+  HistoryBlocker,
+  InvariantRoute,
+  InvariantRoutes,
+  LinkProps,
+  Location,
+  LocationShape,
+  Match,
+  MatchedInvariantRoute,
+  MatchedRoute,
+  MatchParams,
+  MemoryRouterProps,
+  Navigation,
+  NavigationStatics,
+  NavigationType,
+  Query,
+  ResourceStoreContext,
+  ResourceStoreData,
+  Route,
+  RouteContext,
+  RouterAction,
+  RouterActionPush,
+  RouterActionReplace,
+  RouterActionsType,
+  RouteResource,
+  RouteResourceData,
+  RouteResourceError,
+  RouteResourceResponse,
+  RouterProps,
+  RouterState,
+  RouterStoreContext,
+  RouterSubscriberProps,
+  Routes,
+  StaticRouterProps,
+} from './types.js.flow';
+
+export {
+  HistoryAction,
+  HistoryBlocker,
+  InvariantRoutes,
+  LinkProps,
+  Location,
+  LocationShape,
+  Match,
+  MatchedRoute,
+  MatchParams,
+  Navigation,
+  NavigationStatics,
+  NavigationType,
+  Query,
+  ResourceStoreData,
+  Route,
+  RouteContext,
+  RouterAction,
+  RouterActionPush,
+  RouterActionReplace,
+  RouterActionsType,
+  RouteResource,
+  RouteResourceError,
+  RouteResourceResponse,
+  RouterProps,
+  RouterStoreContext,
+  RouterSubscriberProps,
+  Routes,
+} from './types.js.flow';
```

Comment:
You shouldn't need to specify the full path here and in the other files. `./types` should be sufficient since flow will match the correct file for you. I would also make sure that you integrate this in Jira and make sure the types work as expected.

Reply:
I've corrected this and tested it in Jira frontend without issue